src/resources/filters/customnodes/floatreftarget.lua

```diff
@@ -787,9 +787,44 @@ end, function(float)
       float.identifier)
   end
 
-  float.caption_long.content:insert(1, pandoc.RawInline("asciidoc", ". "))
+  if float.type == "Table" and float.content.t == "Table" then
+    -- special-case the situation where the figure is Table and the content is Table
+    --
+    -- just return the table itself with the caption inside the table
+    float.content.caption.long = float.caption_long
+    float.content.attr = pandoc.Attr(float.identifier, float.classes or {}, float.attributes or {})
+    return pandoc.Blocks({
+      pandoc.RawBlock("asciidoc", "[[" .. float.identifier .. "]]\n"),
+      float.content
+    })
+  end
+
+  if float.type == "Figure" then
+    local count = 0
+    local img
+    _quarto.ast.walk(float.content, {
+      Image = function(node)
+        count = count + 1
+        img = node
+      end
+    })
+    if count == 1 then
+      img.identifier = float.identifier
+      img.caption = quarto.utils.as_inlines(float.caption_long)
+      return pandoc.Figure(
+        {img},
+        {float.caption_long},
+        float.identifier)
+    end
+  end
+
+  quarto.log.output(float)
+  quarto.log.output(float.content)
+
+  -- Fallthrough case, render into a div.
+  float.caption_long.content:insert(1, pandoc.RawInline("asciidoc", "."))
   float.caption_long.content:insert(pandoc.RawInline("asciidoc", "\n[[" .. float.identifier .. "]]\n===="))
-  return pandoc.Div({
+  return pandoc.Blocks({
```

Comment:
I believe there is an issue with this. Simple test for subfigures.

````markdown
---
title: Caption
format: 
  asciidoc: default
  html: default
---

::: {#fig-try}

![Test](surus.png)

![Test2](surus.png)

Hello
:::

````

<details>
<summary>Lua error</summary>

````
Error running filter C:\Users\chris\DOCUME~1\DEV_R\QUARTO~2\src\resources\filters\main.lua:
Block, list of Blocks, or compatible element expected, got table
	while retrieving function argument blocks
	while retrieving arguments for function Blocks
stack traceback:
	...2\src\resources\filters\./customnodes/floatreftarget.lua:825: in field 'render'
	...~1\DEV_R\QUARTO~2\src\resources\filters\./ast/render.lua:45: in local 'filter_fn'
	...V_R\QUARTO~2\src\resources\filters\./ast/customnodes.lua:164: in function <...V_R\QUARTO~2\src\resources\filters\./ast/customnodes.lua:154>
	(...tail calls...)
	[C]: in ?
	[C]: in method 'walk'
	...V_R\QUARTO~2\src\resources\filters\./ast/customnodes.lua:86: in function <...V_R\QUARTO~2\src\resources\filters\./ast/customnodes.lua:76>
	(...tail calls...)
	..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:82: in local 'callback'
	..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:97: in upvalue 'run_emulated_filter_chain'
	..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:133: in function <..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:130>
stack traceback:
	...V_R\QUARTO~2\src\resources\filters\./ast/customnodes.lua:86: in function <...V_R\QUARTO~2\src\resources\filters\./ast/customnodes.lua:76>
	(...tail calls...)
	..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:82: in local 'callback'
	..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:97: in upvalue 'run_emulated_filter_chain'
	..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:133: in function <..._R\QUARTO~2\src\resources\filters\./ast/runemulation.lua:130>
ERROR: Error
    at renderFiles (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/render/render-files.ts:346:23)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async renderProject (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/render/project.ts:268:23)
    at async renderForPreview (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/preview/preview.ts:436:24)
    at async render (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/preview/preview.ts:167:22)
    at async preview (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/preview/preview.ts:184:18)
    at async Command.fn (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/preview/cmd.ts:415:7)
    at async Command.execute (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/vendor/deno.land/x/cliffy@v0.25.4/command/command.ts:1790:7)
    at async quarto (file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/quarto.ts:155:3)
    at async file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/quarto.ts:186:5
````

</details>

Possibly because in 
````
{ 
   float.caption_long,
    -- pandoc.RawBlock("asciidoc", "[[" .. float.identifier .. "]]\n====\n"),
    float.content,
    pandoc.RawBlock("asciidoc", "====\n\n")
}
````

`float.content` is already of type `Blocks`, and you can't nest 

````
> pandoc lua -i
Lua 5.4.4  Copyright (C) 1994-2022 Lua.org, PUC-Rio
Embedded in pandoc 3.1.11
> pandoc.Blocks({pandoc.Para("Test"), pandoc.Blocks({pandoc.Para("nested")})})
Block, list of Blocks, or compatible element expected, got table
        while retrieving function argument blocks
        while retrieving arguments for function Blocks
stack traceback:
        stdin:1: in main chunk
````

BTW this is not an issue with this PR only. Using `pandoc.Div` is the same 

````
Error running filter C:\Users\chris\scoop\apps\quarto-prerelease\current\share\filters\main.lua:
Block, list of Blocks, or compatible element expected, got table
        while retrieving function argument content
        while retrieving arguments for function Div
````

Reply:
> Asciidoc output probably needs to be check thoroughly for all features and probably validate or else.

We don't have a validator available, but we do have a number of asciidoc tests in `docs/smoke-all/crossrefs/float/asciidoc`.